src/lightning_habana/fabric/plugins/precision.py

```diff
@@ -12,66 +12,53 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-from typing import Literal, Optional, Union, cast
+from contextlib import contextmanager
+from typing import Generator, Literal
 
+import torch
 from lightning_utilities import module_available
 from typing_extensions import get_args
 
 if module_available("lightning"):
     from lightning.fabric.plugins.precision.precision import Precision
-    from lightning.fabric.utilities.rank_zero import rank_zero_info
 elif module_available("pytorch_lightning"):
     from lightning_fabric.plugins.precision.precision import Precision
-    from lightning_fabric.utilities.rank_zero import rank_zero_info
 else:
     raise ModuleNotFoundError("You are missing `lightning` or `pytorch-lightning` package, please install it.")
 
 
-from lightning_habana import _HPU_AVAILABLE
-from lightning_habana.utils.imports import _HABANA_FRAMEWORK_AVAILABLE
-
-if _HABANA_FRAMEWORK_AVAILABLE:
-    from habana_frameworks.torch.hpex import hmp
-
-_PRECISION_INPUT_INT = Literal[32]
-_PRECISION_INPUT_STR = Literal["32", "bf16", "32-true", "bf16-mixed"]
-_PRECISION_INPUT = Union[_PRECISION_INPUT_INT, _PRECISION_INPUT_STR]
+_PRECISION_INPUT = Literal["32", "bf16", "32-true", "bf16-mixed"]
 
 
 class HPUPrecision(Precision):
     """Plugin that enables bfloat support on HPUs.
 
     Args:
-        precision: The precision to use.
-        opt_level: Choose optimization level for hmp.
-        bf16_file_path: Path to bf16 ops list in hmp O1 mode.
-        fp32_file_path: Path to fp32 ops list in hmp O1 mode.
-        verbose: Enable verbose mode for hmp.
+        precision: to enable ``torch.bfloat16`` (``'bf16-mixed'``).
+        device: The device for ``torch.autocast``.
     """
 
     def __init__(
         self,
         precision: _PRECISION_INPUT,
-        opt_level: str = "O2",
-        bf16_file_path: Optional[str] = None,
-        fp32_file_path: Optional[str] = None,
-        verbose: bool = False,
+        device: str = "hpu",
     ) -> None:
-        rank_zero_info(
```

Comment:
why is this already removed?

Reply:
From 1.11.0 synpase AI release, this is supported fully. Asd we have already changed the default docker image, its removed

Reply:
updated logs